### PR TITLE
wgc

### PR DESCRIPTION
**Pull request by aep**
done

### CODE REVIEW

**Comment by aep** [comment]
@Karitham have a looky

File: wgc.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+				addr = &net.IPNet{
+					IP:   ip,
+					Mask: net.CIDRMask(32, 32),
+				}
+			} else {
+				addr = &net.IPNet{
+					IP:   ip,
+					Mask: net.CIDRMask(128, 128),
+				}
+			}
```
Comment (Karitham) [inline wgc.go:133]:
```suggestion
	        addr = &net.IPNet{
                    IP: ip,
                    Mask: mask(ip),
                }
```

**Review by Karitham — CHANGES_REQUESTED** [review]

File: apis/wga.kraudcloud.com/v1beta/types.go [hunk 2]
```diff
@@ -47,3 +47,38 @@ type WireguardAccessPeerStatusPeer struct {
...
+type WireguardClusterClient struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              WireguardClusterClientSpec    `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status            *WireguardClusterClientStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+type WireguardClusterClientSpec struct {
+	Address             string                               `yaml:"address" json:"address"`
+	PrivateKey          WireguardClusterClientSpecPrivateKey `yaml:"privateKey" json:"privateKey"`
```
[the remark this answers is not in the document]
Comment (aep) [reply apis/wga.kraudcloud.com/v1beta/types.go:64]: oh yeah, so the way it works tho is that you can actually add a verbatim key in privateKey.value.
wgc will then create the secret and remove the value.

still think it should be renamed?

File: charts/wga/templates/wgc-crds.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+              properties:
+                publicKey:
+                  type: string
+            spec:
+              type: object
+              properties:
+                address:
+                  type: string
+                  description: local inner ip address
+                privateKey:
```
[the remark this answers is not in the document]
Comment (aep) [reply charts/wga/templates/wgc-crds.yaml:40]: what, thats the same key, but you said different things

File: charts/wga/templates/daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+              memory: "128Mi"
+              cpu: "500m"
+            limits:
+              memory: "128Mi"
+              cpu: "500m"
+            - name: GOMEMLIMIT
+              valueFrom:
+                resourceFieldRef:
+                  containerName: wgc
+                  resource: limits.memory
```
[the remark this answers is not in the document]
Comment (aep) [reply charts/wga/templates/daemonset.yaml:34]: wtf yes, why did this even work

File: crds.go [hunk 5]
```diff
@@ -64,21 +68,151 @@ func epMain(
...
+					if err != nil {
+						slog.Error("Error generating key", "error", err)
+						return nil, fmt.Errorf("error generating key: %w", err)
+					}
+				}
+
+				sk.Name = skName
+				sk.Namespace = skNamespace
+				sk.Data = make(map[string][]byte)
+				sk.Data[skKey] = []byte(privk.String())
```
[the remark this answers is not in the document]
Comment (aep) [reply crds.go:138]: no thats wrong

File: peers.go [hunk 6]
```diff
@@ -173,6 +175,7 @@ func fmtPeer(peer v1beta.WireguardAccessPeer, dns []net.IP, pk, psk wgtypes.Key)
 
 	oubuf := &strings.Builder{}
 	err := Format(oubuf, ConfigFile{
```
[the remark this answers is not in the document]
Comment (aep) [reply peers.go:177]: too lazy

File: charts/wga/templates/wgc-crds.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (aep) [reply charts/wga/templates/wgc-crds.yaml:40]: never mind, i get your idea

**Pull request merged**